src/lib/yang/binary.lua

```diff
@@ -210,11 +210,19 @@ local function data_emitter(production)
          end
       end
    end
+   local native_types = {
+      ["instance-identifier"] = true,
+      empty = true,
```

Comment:
Correct, it will have no data. Point is `empty` is implemented as a string-like datatype (it lacks ctype). That was the main reason I added to this list of "native-types". If I removed from this list, it will tried to be compiled as a ctype value and since it lacks one the code will fail.

I wrote down a test that reads a schema instance featuring an empty node and writes it down to a binary file. It's not working. I got the following error:

```
lib/yang/binary.lua:53: attempt to index a boolean value
stack traceback:
        core/main.lua:149: in function '__index'
        lib/yang/binary.lua:53: in function 'emit'
        lib/yang/binary.lua:259: in function 'compile_data_for_schema'
        test.lua:26: in function 'selftest'
        program/snsh/snsh.lua:31: in function <program/snsh/snsh.lua:31>
        core/lib.lua:434: in function 'dogetopt'
        program/snsh/snsh.lua:65: in function 'run'
        core/main.lua:67: in function <core/main.lua:43>
        [C]: in function 'xpcall'
        core/main.lua:211: in main chunk
        [C]: at 0x55b4bb2c2f30
        [C]: in function 'pcall'
        core/startup.lua:3: in main chunk
        [C]: in function 'require'
        [string "require "core.startup""]:1: in main chunk
```

This is the test:

```lua
module(..., package.seeall)

local binary = require("lib.yang.binary")
local data = require("lib.yang.data")
local schema = require("lib.yang.schema")

local compile_data_for_schema = binary.compile_data_for_schema

function selftest (args)
   local test_schema = schema.load_schema([[module test {
      namespace "test";
      prefix "test";

      container foo {
         leaf enable-qos {
            type empty;
         }
      }
   }]])
   local data = data.load_data_for_schema(test_schema, [[
      foo {
         enable-qos;
      }
   ]])
   local tmp = os.tmpname()
   compile_data_for_schema(test_schema, data, tmp)
end
```

The test gets fixed if empty.tostring returns "true", instead of an empty string. It also made me think whether an empty type should be compiled as ctype bool that always returns true. I think this makes more sense. Thoughts?